lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2017,6 +2025,25 @@ formGraphOp(SILTensorOpInfo &opInfo,
         return diagnoseInvalidAttr("may not use this device name");
     }
 
+    auto verifyShapeAttr = [&](SymbolicValue constValue) -> bool {
+      // strip-away the possible aggregate wrapper.
+      constValue = constValue.lookThroughSingleElementAggregates();
+      if (constValue.getKind() != SymbolicValue::Array) {
+        constValue.dump();
+        diagnoseInvalidAttr("requires an array");
+        return true;
+      }
+      auto elements = constValue.getArrayValue();
+      for (auto elt : elements) {
+        // strip-away the possible aggregate wrapper.
+        elt = elt.lookThroughSingleElementAggregates();
+        if (elt.getKind() != SymbolicValue::Integer) {
+          diagnoseInvalidAttr("Shape attr requires an array of ints!");
```

Comment:
diagnostics should be lower case, and not include explanation marks  :-)

Reply:
Good catch! I was confusing with an assert code path again. Removed "Shape attr" to be consistent with other similar error messages.

For future reference, when we catch a shape error at compile time, I think that'd deserve some serious exclamation marks!!!! :)